question25/main.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+const textLinkToTerms = document.getElementById('js-terms-textlink');
+const modal = document.getElementById('js-modal');
+const modalClose = document.getElementById('js-close');
+const mask = document.getElementById('js-mask');
+const checkbox = document.getElementById('js-checkbox');
+const submitButton = document.getElementById('js-submit-button');
+const modalContainer = document.getElementById("js-modal-container");
+const name = document.getElementById('name');
+const mail = document.getElementById('mail');
+const password = document.getElementById('password');
+
+
+textLinkToTerms.addEventListener('click',()=>{
+    modal.classList.remove('hidden');
+    mask.classList.remove('hidden');
+});
+modalClose.addEventListener('click',()=>{
+    modal.classList.add('hidden');
+    mask.classList.add('hidden');
+});
+mask.addEventListener('click',()=>{
+    modalClose.click();
+});
+
+const options = {
+  root: document.getElementById('js-modal'),
+  threshold: 1
+};
+
+const checkWhenIntersect = ([entry]) => {
+    if (entry.isIntersecting) {
+      checkbox.checked = true;
+      checkbox.disabled = false;
+      switchSubmitButton(); 
+    }
+}
```

Comment:
IMO
It is a matter of preference; it is simpler to write than an if statement. No need to fix it.
```suggestion
const checkWhenIntersect = ([entry]) => {
  checkbox.checked = entry.isIntersecting;
  checkbox.disabled = !entry.isIntersecting;
  entry.isIntersecting && switchSubmitButton();
};
```

Reply:
Thank you for your review! 
The idea you presented is cleaner and has been revised. Thank you for your instruction, I learned a lot！

fixed commit : https://github.com/csakyo/homework/pull/39/commits/f0de54dddc7783d37ba7887e9b1d417f0e0c5e5a